### PR TITLE
Update bundle metadata for AMQ Streams 1.7.0 RC1

### PR DESCRIPTION
**Pull request by kyguy**
To make the backporting process easier, I

- renamed the CRD file names to match upstream
- renamed the CSV file to `bundle.clusterserviceversion.yaml`

Neither of these changes affect how OLM installs AMQ Streams 1.7.0

(P.S. MD5 SHA update will come from a later PR, once the images are baked)

### CODE REVIEW

File: operator-metadata/manifests/bundle.clusterserviceversion.yaml [hunk 1]
```diff
@@ -1284,5 +763,516 @@ spec:
...
+
+    * The ability to partition topics for horizontal scalability
+
+    ### CRD Upgrades
+
+    **!!! IMPORTANT !!!** 
+    This release introduces new API version `v1beta2` to all Strimzi custom resources. 
+    This is a preparation for migration to `apiextensions/v1` which is needed because Kubernetes 1.22 will remove support for `apiextensions/v1beta1`. 
+    Migration to `v1beta2` needs to be completed for all Strimzi CRDs and CRs after the upgrade to 0.22 is done and before upgrading to Strimzi 0.23 which will support only Strimzi `v1beta2` APIs and `apiextensions/v1` CRDs.
+    For more details about the CRD upgrades, see the [documentation](https://strimzi.io/docs/operators/0.22.1/deploying.html#assembly-upgrade-resources-str).
```
Comment (scholzj) [inline operator-metadata/manifests/bundle.clusterserviceversion.yaml:787]: This has to link to downstream docs (but 1.7 downstream docs - we screwed this last time). Ask Paul and Daniel what will be the URL for this.

File: operator-metadata/manifests/bundle.clusterserviceversion.yaml [hunk 2]
```diff
@@ -1284,5 +763,516 @@ spec:
...
+
+    ### What is new in AMQ Streams 1.7
+
+    * Add v1beta2 version for all resources. v1beta2 removes all deprecated fields.
+
+    * Add annotations that enable the operator to restart Kafka Connect connectors or tasks. The annotations can be applied to the KafkaConnector and the KafkaMirrorMaker2 custom resources.
+
+    * Add additional configuration options for the Kaniko executor used by the Kafka Connect Build on Kubernetes
+
+    * Add support for JMX options configuration of all Kafka Connect (KC, KC2SI, MM2)
```
Comment (scholzj) [inline operator-metadata/manifests/bundle.clusterserviceversion.yaml:797]: I would remove this.

File: operator-metadata/manifests/bundle.clusterserviceversion.yaml [hunk 3]
```diff
@@ -1284,5 +763,516 @@ spec:
...
+
+    * Update Strimzi Kafka OAuth to version 0.7 and add support for new features (OAuth over SASL PLAIN, Checking token audience, Validating tokens using JSONPath)
+
+    * Configure external logging ConfigMap name and key.
+
+    * Add support for configuring labels and annotations in ClusterRoleBindings created as part of Kafka and Kafka Connect clusters
+
+    * Add support for Ingress v1 in Kubernetes 1.19 and newer
+
+    * Add support for Kafka 2.6.1
```
Comment (scholzj) [inline operator-metadata/manifests/bundle.clusterserviceversion.yaml:807]: This should be removed.

File: operator-metadata/manifests/bundle.clusterserviceversion.yaml [hunk 4]
```diff
@@ -1284,5 +763,516 @@ spec:
...
+
+    * List topics used by a Kafka Connect connector in the .status section of the KafkaConnector custom resource
+
+    * Bump Cruise Control to v2.5.37 for Kafka 2.7 support. Note this new version of Cruise Control uses Log4j 2 and is supported by dynamic logging configuration (where logging properties are defined in a ConfigMap). However, existing Log4j configurations must be updated to Log4j 2 configurations.
+
+    * Support pausing reconciliation of CR with annotation strimzi.io/pause-reconciliation
```
Comment (scholzj) [inline operator-metadata/manifests/bundle.clusterserviceversion.yaml:813]: Maybe we should remove this whole section. It has two problems:
* It covers things AMQ Streams do nto support (Ingress, Kafka 2.6.1, etc.)
* It covers only 0.22. Not 0.21 features. But AMQ Streams 1.7 has to include both. So if we want to keep this, we should extend this with the new features from 0.21 as well.

File: operator-metadata/manifests/bundle.clusterserviceversion.yaml [hunk 5]
```diff
@@ -1284,5 +763,516 @@ spec:
...
+
+    * Add support for JMX options configuration of all Kafka Connect (KC, KC2SI, MM2)
+
+    * Update Strimzi Kafka OAuth to version 0.7 and add support for new features (OAuth over SASL PLAIN, Checking token audience, Validating tokens using JSONPath)
+
+    * Configure external logging ConfigMap name and key.
+
+    * Add support for configuring labels and annotations in ClusterRoleBindings created as part of Kafka and Kafka Connect clusters
+
+    * Add support for Ingress v1 in Kubernetes 1.19 and newer
```
Comment (scholzj) [inline operator-metadata/manifests/bundle.clusterserviceversion.yaml:805]: Thsi should be removed.

File: operator-metadata/manifests/bundle.clusterserviceversion.yaml [hunk 6]
```diff
@@ -1284,5 +763,516 @@ spec:
...
+    Migration to `v1beta2` needs to be completed for all Strimzi CRDs and CRs after the upgrade to 0.22 is done and before upgrading to Strimzi 0.23 which will support only Strimzi `v1beta2` APIs and `apiextensions/v1` CRDs.
+    For more details about the CRD upgrades, see the [documentation](https://strimzi.io/docs/operators/0.22.1/deploying.html#assembly-upgrade-resources-str).
+
+    ### What is new in AMQ Streams 1.7
+
+    * Add v1beta2 version for all resources. v1beta2 removes all deprecated fields.
+
+    * Add annotations that enable the operator to restart Kafka Connect connectors or tasks. The annotations can be applied to the KafkaConnector and the KafkaMirrorMaker2 custom resources.
+
+    * Add additional configuration options for the Kaniko executor used by the Kafka Connect Build on Kubernetes
```
Comment (scholzj) [inline operator-metadata/manifests/bundle.clusterserviceversion.yaml:795]: Thsi should be removed.

File: operator-metadata/manifests/bundle.clusterserviceversion.yaml [hunk 7]
```diff
@@ -1284,5 +763,516 @@ spec:
...
 
+    * **Cluster Rebalancing** - Uses built-in Cruise Control for redistributes partition replicas according to specified goals in order to achieve the best cluster performance.
+
+    * **Monitoring** - Built-in support for monitoring using Prometheus and provided Grafana dashabords
+
+    ### Upgrading your Clusters
+    
+    The Strimzi Operator understands how to run and upgrade between a set of Kafka versions.
+    When specifying a new version in your config, check to make sure you aren't using any features that may have been removed.
+    See [the upgrade guide](https://strimzi.io/docs/operators/latest/deploying.html#assembly-upgrading-kafka-versions-str) for more information.
```
Comment (scholzj) [inline operator-metadata/manifests/bundle.clusterserviceversion.yaml:839]: Again, this needs to point to downstream docs.

File: operator-metadata/manifests/bundle.clusterserviceversion.yaml [hunk 8]
```diff
@@ -1284,5 +763,516 @@ spec:
...
+
+    ### Storage
+
+    An efficient data storage infrastructure is essential to the optimal performance of Apache Kafka®.
+    Apache Kafka® deployed via Strimzi requires block storage.
```
Comment (scholzj) [inline operator-metadata/manifests/bundle.clusterserviceversion.yaml:844]:
```suggestion
    Apache Kafka® deployed via AMQ Streams requires block storage.
```

File: operator-metadata/manifests/bundle.clusterserviceversion.yaml [hunk 9]
```diff
@@ -1284,5 +763,516 @@ spec:
...
+
+    * **Includes HTTP Kafka Bridge** - Allows clients to send and receive messages through an Apache Kafka® cluster via HTTP protocol.
 
+    * **Cluster Rebalancing** - Uses built-in Cruise Control for redistributes partition replicas according to specified goals in order to achieve the best cluster performance.
+
+    * **Monitoring** - Built-in support for monitoring using Prometheus and provided Grafana dashabords
+
+    ### Upgrading your Clusters
+    
+    The Strimzi Operator understands how to run and upgrade between a set of Kafka versions.
```
Comment (scholzj) [inline operator-metadata/manifests/bundle.clusterserviceversion.yaml:837]:
```suggestion
    The AMQ Streams Operator understands how to run and upgrade between a set of Kafka versions.
```

File: operator-metadata/manifests/bundle.clusterserviceversion.yaml [hunk 10]
```diff
@@ -1284,5 +763,516 @@ spec:
...
+    
+    * Persistent
+    
+    * JBOD (Just a Bunch of Disks, suitable for Kafka only. Not supported in Zookeeper.)
+    
+    Strimzi also supports advanced operations such as adding or removing disks in Apache Kafka® brokers or resizing the persistent volumes (where supported by the infrastructure).
+
+    ### Documentation
+
+    Documentation for the current release can be found on Red Hat [website](https://access.redhat.com/documentation/en-us/red_hat_amq/2020.q4/html-single/using_amq_streams_on_openshift/index).
```
Comment (scholzj) [inline operator-metadata/manifests/bundle.clusterserviceversion.yaml:859]: We need to use the 1.7 URL.

File: operator-metadata/manifests/bundle.clusterserviceversion.yaml [hunk 11]
```diff
@@ -1284,5 +763,516 @@ spec:
...
+                strimzi.io/kind: cluster-operator
+            spec:
+              serviceAccountName: strimzi-cluster-operator
+              volumes:
+                - name: strimzi-tmp
+                  emptyDir:
+                    medium: Memory
+                - name: co-config-volume
+                  configMap:
+                    name: strimzi-cluster-operator
```
Comment (scholzj) [inline operator-metadata/manifests/bundle.clusterserviceversion.yaml:1159]: You seem to have changed the name. Which name is used in the docs? Is the change worth the possible confusion for users?

File: operator-metadata/manifests/bundle.clusterserviceversion.yaml [hunk 12]
```diff
@@ -1284,5 +763,516 @@ spec:
...
+                    - containerPort: 8080
+                      name: http
+                  args:
+                    - /opt/strimzi/bin/cluster_operator_run.sh
+                  volumeMounts:
+                    - name: strimzi-tmp
+                      mountPath: /tmp
+                    - name: co-config-volume
+                      mountPath: /opt/strimzi/custom-config/
+                  env:
```
Comment (scholzj) [inline operator-metadata/manifests/bundle.clusterserviceversion.yaml:1173]: You lost the 3scale annotations and labels.

File: operator-metadata/manifests/bundle.clusterserviceversion.yaml [hunk 13]
```diff
@@ -1284,5 +763,516 @@ spec:
...
+    - data-streams
+    - streaming
+    - streams
+  labels:
+    name: strimzi-cluster-operator
+  links:
+    - name: Product Page
+      url: https://access.redhat.com/products/red-hat-amq
+    - name: Documentation
+      url: https://access.redhat.com/documentation/en-us/red_hat_amq/2020.q4/html-single/using_amq_streams_on_openshift/index
```
Comment (scholzj) [inline operator-metadata/manifests/bundle.clusterserviceversion.yaml:1267]: Needs to be updated to the 1.7 link.

File: operator-metadata/manifests/bundle.clusterserviceversion.yaml [hunk 14]
```diff
@@ -1284,5 +763,516 @@ spec:
...
+  maintainers:
+    - email: customerservice@redhat.com
+      name: Red Hat
+  maturity: stable
+  provider:
+    name: Red Hat
+  replaces: amqstreams.v1.6.3
+  selector:
+    matchLabels:
+      name: strimzi-cluster-operator
```
Comment (scholzj) [inline operator-metadata/manifests/bundle.clusterserviceversion.yaml:1277]: This selector changed from 1.6 ... I think this could be an issue?

File: operator-metadata/manifests/bundle.clusterserviceversion.yaml (hunk 1 above)
Comment (kyguy) [reply operator-metadata/manifests/bundle.clusterserviceversion.yaml:787]: @laidan6000 Do you know what downstream link we should put here?

File: operator-metadata/manifests/bundle.clusterserviceversion.yaml (hunk 10 above)
Comment (kyguy) [reply operator-metadata/manifests/bundle.clusterserviceversion.yaml:859]: @laidan6000 Do you know what downstream link we should put here?

File: operator-metadata/manifests/bundle.clusterserviceversion.yaml (hunk 13 above)
Comment (kyguy) [reply operator-metadata/manifests/bundle.clusterserviceversion.yaml:1267]: @laidan6000 Do you know what downstream link we should put here?

File: operator-metadata/manifests/bundle.clusterserviceversion.yaml (hunk 4 above)
Comment (kyguy) [reply operator-metadata/manifests/bundle.clusterserviceversion.yaml]: I removed this altogether to avoid including incorrect information.

File: operator-metadata/manifests/bundle.clusterserviceversion.yaml (hunk 11 above)
Comment (kyguy) [reply operator-metadata/manifests/bundle.clusterserviceversion.yaml]: I changed it back to the original name for 1.7.0 as I don't know what the impact would be for changing it. I'll look into aligning it with upstream next release

File: operator-metadata/manifests/bundle.clusterserviceversion.yaml (hunk 12 above)
Comment (kyguy) [reply operator-metadata/manifests/bundle.clusterserviceversion.yaml:1147]: Added them back in

File: operator-metadata/manifests/bundle.clusterserviceversion.yaml (hunk 14 above)
Comment (kyguy) [reply operator-metadata/manifests/bundle.clusterserviceversion.yaml]: Changed them back to their original name

File: operator-metadata/manifests/bundle.clusterserviceversion.yaml (hunk 1 above)
Comment (laidan6000) [reply operator-metadata/manifests/bundle.clusterserviceversion.yaml:787]: https://access.redhat.com/documentation/en-us/red_hat_amq/2021.q2/html/deploying_and_upgrading_amq_streams_on_openshift/assembly-upgrade-str#assembly-upgrade-resources-str

File: operator-metadata/manifests/bundle.clusterserviceversion.yaml (hunk 7 above)
Comment (laidan6000) [reply operator-metadata/manifests/bundle.clusterserviceversion.yaml:813]: https://access.redhat.com/documentation/en-us/red_hat_amq/2021.q2/html/deploying_and_upgrading_amq_streams_on_openshift/assembly-upgrade-str

File: operator-metadata/manifests/bundle.clusterserviceversion.yaml (hunk 10 above)
Comment (laidan6000) [reply operator-metadata/manifests/bundle.clusterserviceversion.yaml:833]: https://access.redhat.com/documentation/en-us/red_hat_amq/2021.q2/html/using_amq_streams_on_openshift/index

File: operator-metadata/manifests/bundle.clusterserviceversion.yaml (hunk 13 above)
Comment (laidan6000) [reply operator-metadata/manifests/bundle.clusterserviceversion.yaml:1250]: https://access.redhat.com/documentation/en-us/red_hat_amq/2021.q2/html/using_amq_streams_on_openshift/index

File: operator-metadata/manifests/bundle.clusterserviceversion.yaml [hunk 15]
```diff
@@ -1284,5 +763,499 @@ spec:
...
+
+    * The ability for a consumer to replay streams of events
+
+    * The ability to partition topics for horizontal scalability
+
+    ### CRD Upgrades
+
+    **!!! IMPORTANT !!!** 
+    This release introduces new API version `v1beta2` to all AMQ Streams custom resources.
+    This is a preparation for migration to `apiextensions/v1` which is needed because Kubernetes 1.22 will remove support for `apiextensions/v1beta1`.
```
Comment (scholzj) [inline operator-metadata/manifests/bundle.clusterserviceversion.yaml:785]:
```suggestion
    This is a preparation for migration to `apiextensions/v1` which is needed because OpenShift 4.9 will remove support for `apiextensions/v1beta1`. 
```

File: operator-metadata/manifests/strimziclusteroperator.configmap.yaml [hunk 16]
```diff
@@ -1,7 +1,7 @@
 kind: ConfigMap
 apiVersion: v1
 metadata:
-  name: amq-streams-cluster-operator
+  name: strimzi-cluster-operator
```
Comment (scholzj) [inline operator-metadata/manifests/strimziclusteroperator.configmap.yaml:4]: I think you need to revert this change to match what you have in the deployment.

**Review by scholzj — APPROVED** [review]
Two more comments. LGTM otherwise
